python/pyiceberg/io/pyarrow.py

```diff
@@ -66,10 +74,14 @@ class PyArrowFile(InputFile, OutputFile):
         >>> # output_file.create().write(b'foobytes')
     """
 
-    def __init__(self, location: str, path: str, fs: FileSystem):
+    _normalized_location: NormalizedLocation
+
+    def __init__(self, location: Union[str, NormalizedLocation], fs: FileSystem):
```

Comment:
This works, but I think it is probably cleaner to pass both `location` and `path` into `PyArrowFile` like it was before.

The problem is that this needs to re-create the location that was passed in to create this, rather than just passing the path directly. It also makes a few changes in this class that are unnecessary if `PyArrowFileIO` passes location, path, and fs.

Reply:
Got it, I've reverted the bit and passed in the original location. I think PyArrow itself should be responsible for handling the URL. Otherwise, users need to construct the path when they directly initialize the PyArrow Input or Output file. Let me know what you think of the current approach